sdks/python/apache_beam/metrics/base.py

```diff
@@ -0,0 +1,79 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+
+Available classes:
+- Metric - Base class of a metrics object. Provides very basic methods.
```

Comment:
Why is the `Metric` class visible. It seems like users won't be able to do anything useful with it. If its only visible so methods can be called, etc. maybe make it clear that none of these are intended to be sub-classed, rather they are created with the factory methods? And then document how to create each one.